index_mh_iter.go

```diff
@@ -9,13 +9,13 @@ import (
 	"github.com/multiformats/go-multihash"
 )
 
-var _ MultihashIterator = (*indexMhIterator)(nil)
+var _ MultihashIterator = (*MhIterator)(nil)
 
-type indexMhIterator struct {
-	steps []iteratorStep
-
-	curStep    int
-	lastOffset uint64
+// MhIterator is a simple MultihashIterator implementation that iterates a
+// slice of multihash.Multihash.
+type MhIterator struct {
```

Comment:
* I am curious why the PR exports this type? 
* Would it make sense to name it `SliceMultihashIterator` or something along those lines please? Having both `MultihashIterator` and `MhIterator` could be confusing to the user of the API.
  * We can then replace other utilitarian implementations like [this](https://github.com/filecoin-project/index-provider/blob/main/engine/slicemhiter_test.go).

Reply:
It was exported for _potential_ future use like in #226, so maybe best left unexported here and make that change in such future PR.

It would be better to call it `sliceMultihashIterator`.